### PR TITLE
[docs] #161: Add more info about ISI

### LINKED ISSUE #161: Update the list of Iroha Special Instructions
image

https://hyperledger.github.io/iroha-2-docs/guide/advanced/isi.html

> actually, this really needs to be in the docs
> people reading the docs want to have a nice summary of all instructions on one page, not look through many files in rust
> ideally in a nice table that groups concepts together

We need to review the ISI chapter and make sure we list all the instructions on one page.
This is connected to #85 as well since at least some of this info should be in the chapter about objects

### PR DESCRIPTION
**Pull request by outoftardis**
Signed-off-by: Ekaterina Mekhnetsova <mekkatya@gmail.com>

Closes #161

### CODE REVIEW

File: src/guide/advanced/intro.md [hunk 1]
```diff
@@ -7,3 +7,69 @@ Here we cover in some breadth the [Iroha special instructions](./isi.md),
...
+The reasoning behind this algorithm is simple: if someone had some evil
+peers and connected them to the existing network, if they tried to fake
+data, some good™ peers would not get the same (evil™) world state. If
+that's the case, the evil™ peers would not be allowed to participate in
+consensus, and you would eventually produce a block using only good™ peers.
+
+As a natural consequence, if any changes to the world state are made
+without the use of ISI, the good™ peers cannot know of them. They won't be
+able to reproduce the hash of the world state, and thus consensus will
+fail. Same thing happens if the peers have different instructions.
```
Comment (WRRicht3r) [inline src/guide/advanced/intro.md:70]:
```suggestion
fail. The same thing happens if the peers have different instructions.
```

File: src/guide/advanced/isi.md [hunk 2]
```diff
@@ -1,77 +1,29 @@
...
-
-As a natural consequence, if any changes to the world state are made
-without the use of ISI, the good™ peers cannot know of them. They won't be
-able to reproduce the hash of the world state, and thus consensus will
-fail. Same thing happens if the peers have different instructions.
-
-## Instructions
-
-So what kinds of special instructions do we have? If you've read the
+When we talked about [how Iorha operates](intro.md#how-iroha-works), we
```
Comment (WRRicht3r) [inline src/guide/advanced/isi.md:3]:
```suggestion
When we spoke about [how Iroha operates](intro.md#how-iroha-works), we
```

File: src/guide/advanced/isi.md [hunk 3]
```diff
@@ -1,77 +1,29 @@
...
+said that Iroha Special Instructions is the only way to modify the world
```
Comment (WRRicht3r) [inline src/guide/advanced/isi.md:4]:
```suggestion
said that Iroha Special Instructions are the only way to modify the world
```

File: src/guide/advanced/isi.md [hunk 4]
```diff
@@ -1,77 +1,29 @@
...
+state. So what kinds of special instructions do we have? If you've read the
```
Comment (WRRicht3r) [inline src/guide/advanced/isi.md:5]:
```suggestion
state. So, what kind of special instructions do we have? If you've read the
```

File: src/guide/advanced/isi.md [hunk 5]
```diff
@@ -1,77 +1,29 @@
...
 tutorial for [Rust](../rust.md) or [Python](../python.md), you've already
 seen a couple of instructions: `Register<Account>` and `Mint<Quantity>`.
 
-For the exhaustive list of Instructions you should consult
-[our source code](https://github.com/hyperledger/iroha/tree/iroha2-dev/core/src/smartcontracts/isi).
+In the chapter on blockchain objects, we provide you with
```
Comment (WRRicht3r) [inline src/guide/advanced/isi.md:9]:
```suggestion
In the chapter on blockchain objects, we provided you with
```

File: src/guide/advanced/isi.md [hunk 6]
```diff
@@ -1,77 +1,29 @@
...
+[a summary](../objects/instructions.md) of Iroha Special Instructions:
+which objects each instruction can be called for, and what are the
```
Comment (WRRicht3r) [inline src/guide/advanced/isi.md:11]:
```suggestion
what objects each instruction can be called for, and which are the
```

File: src/guide/advanced/isi.md [hunk 7]
```diff
@@ -1,77 +1,29 @@
...
+instructions available for each object.
 
-Here we will cover only some important classes.
+Here we present you with the overview of each instruction.
```
Comment (WRRicht3r) [inline src/guide/advanced/isi.md:14]:
```suggestion
Here we present you with an overview of each instruction.
```

File: src/guide/advanced/isi.md [hunk 8]
```diff
@@ -1,77 +1,29 @@
...
 
-### (Un)Register
+| Instruction                                              | Descriptions                                      |
+| -------------------------------------------------------- | ------------------------------------------------- |
+| [Register/Unregister](#unregister)                       | Give an ID to a new entity on the blockchain.     |
+| [Mint/Burn](#mintburn)                                   | Mint/burn assets, triggers, or permission tokens. |
+| [SetKeyValue/RemoveKeyValue](#setkeyvalueremovekeyvalue) | Update metadata of blockchain objects.            |
+| [Grant/Revoke](#grantrevoke)                             | Give accounts certain permissions or remove them. |
```
Comment (WRRicht3r) [inline src/guide/advanced/isi.md:21]:
```suggestion
| [Grant/Revoke](#grantrevoke)                             | Give or remove certain permissions from accounts. |
```

File: src/guide/advanced/isi.md [hunk 9]
```diff
@@ -131,7 +83,14 @@ usually the one that registered the asset in the first place. All assets
 are assumed to be non-negative as well, so you can never have `-1.0` of
 `time` or `Burn` a negative amount and get a `Mint`.
 
-### Grant/Revoke
+## Transfer
+
+Similarly to mint and burn instructions, transferring refers to assets. You
```
Comment (WRRicht3r) [inline src/guide/advanced/isi.md:88]:
```suggestion
Similar to mint and burn instructions, transferring refers to assets. You
```

File: src/guide/advanced/isi.md [hunk 10]
```diff
@@ -140,47 +99,23 @@ a group of permissions (a "role"). Granted roles and permissions can only
...
-    [Iroha 2 Whitepaper](https://github.com/hyperledger/iroha/blob/iroha2-dev/docs/source/iroha_2_whitepaper.md)
-    is a good start.
+## `ExecuteTrigger`
+
+This instruction is used to execute [triggers](./triggers.md).
+
+## Composite instructions
+
+Iroha also offers composite instructions (`If`, `Pair`, `Sequence`) to
+execute instructions in a certain way:
```
Comment (WRRicht3r) [inline src/guide/advanced/isi.md:116]:
```suggestion
be executed in a certain way:
```

File: src/guide/advanced/isi.md [hunk 11]
```diff
@@ -140,47 +99,23 @@ a group of permissions (a "role"). Granted roles and permissions can only
...
+
+- `If`: execute one of the two given instructions based on a given
+  condition
+- `Sequence`: execute provided vector of instructions in a given order
```
Comment (WRRicht3r) [inline src/guide/advanced/isi.md:120]:
```suggestion
- `Sequence`: execute a provided vector of instructions in a given order
```

**Review by WRRicht3r — CHANGES_REQUESTED** [review]

File: src/guide/advanced/isi.md (hunk 3 above)
Comment (outoftardis) [reply src/guide/advanced/isi.md:4]: isn't it similar to this case?
image
https://webapps.towson.edu/ows/sub-verb.htm

File: src/guide/advanced/isi.md (hunk 6 above)
Comment (outoftardis) [reply src/guide/advanced/isi.md:11]:
```suggestion
what objects each instruction can be called for, and what are the
```

File: src/guide/advanced/isi.md (hunk 4 above)
Comment (outoftardis) [reply src/guide/advanced/isi.md:5]: there are multiple kinds though

File: src/guide/advanced/isi.md (hunk 10 above)
Comment (outoftardis) [reply src/guide/advanced/isi.md:116]: composite instructions execute other instructions in a specified way, so I don't think passive voice is applicable here

File: src/guide/advanced/isi.md (hunk 3 above)
Comment (WRRicht3r) [reply src/guide/advanced/isi.md]: Not really, the instructions are plural, which is why I used the **are**. If it was instruction (singular) then **is** would be more accurate

File: src/guide/advanced/isi.md (hunk 4 above)
Comment (WRRicht3r) [reply src/guide/advanced/isi.md]: yes, that's taken into account as instructions is plural

**Review by WRRicht3r — APPROVED** [review]

**Review by appetrosyan — APPROVED** [review]

**Pull request merged**